src/typecode/language_classifier.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (c) 2018 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from typecode.language_model import score
+import yaml
+import os
+import re
+import io
+
+MAX_LEN = 209   # default sequence lenghth model was trained on'
+
+data_dir = os.path.join(os.path.dirname(__file__), 'data')
+tokens_path = os.path.join(data_dir, "token_indexes.yml")  # used to map words to indexes
+
+with open(tokens_path, 'r') as stream:
+    token_indexes = yaml.safe_load(stream)
+
+classes = ['C', 'C#', 'C++', 'CoffeeScript', 'Common Lisp',
```

Comment:
can you add a comment on what these classes are?

Reply:
It depends upon how the model was trained. the model outputs probabilities, which are indexed in same manner as these classes. for example:

[0.001, 0.012, 0.58, ......]  --> probabilities for ['C', 'C#', 'C++', ......]